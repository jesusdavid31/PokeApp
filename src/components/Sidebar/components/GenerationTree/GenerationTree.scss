.generation-tree {

    .show-all-generations {
        display: flex;
        justify-content: center;
        align-items: center;
        height: 48px;
        background: rgba(0, 0, 0, 0.2);
        backdrop-filter: blur(2px);
        color: #fff;
        padding: 8px 16px;
        cursor: pointer;
        border-left: 4px solid #fff; /* Línea de borde en el lado izquierdo */

        &:hover {
            background: rgba(0, 0, 0, 0.5);
            color: #fff;
        }
    }

    .generation-accordion {
        background: rgba(0, 0, 0, 0.2);
        backdrop-filter: blur(2px);

        .pokemon-list {
            max-height: 300px;
            overflow-y: auto;
            padding: 0px 8px; // Para evitar que el contenido quede oculto detrás del scrollbar

            // Estilos para scroll bonito en navegadores Webkit (Chrome, Edge, Safari)
            &::-webkit-scrollbar {
                width: 8px;
                background: #004A94;
                border-radius: 8px;
            }
            &::-webkit-scrollbar-thumb {
                background: #fff;
                border-radius: 8px;
            }
            &::-webkit-scrollbar-track {
                background: #004A94;
                border-radius: 8px;
            }
        }
    }
}