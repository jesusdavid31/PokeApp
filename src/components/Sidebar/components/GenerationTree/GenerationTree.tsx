/* eslint-disable react-hooks/exhaustive-deps */
import { useEffect } from 'react';

// MUI
import { Box } from '@mui/material';
import Accordion from '@mui/material/Accordion';
import AccordionSummary from '@mui/material/AccordionSummary';
import AccordionDetails from '@mui/material/AccordionDetails';
import Typography from '@mui/material/Typography';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';

// HOOKS, STORE Y UTILIDADES
import { usePokemonGenerationsStore } from '../../../../store/pokemonGenerations.store';
import { usePokemonGenerations } from '../../../../hooks/usePokemonGenerations';

// ESTILOS
import './GenerationTree.scss';

const GenerationTree = () => {

    // HOOK DE GENERACIONES DE POKEMON
    const { 
        loading,
        fetchGenerations
    } = usePokemonGenerations();

    // STORE
    const { generations } = usePokemonGenerationsStore();

    useEffect(() => {
        fetchGenerations();
    }, []);

    return (
        <Box sx={{ width: '100%', marginBottom: 10 }} className="generation-tree">
            {loading ? (
                <Typography>Cargando generaciones...</Typography>
            ): (
                <>
                    {generations.map((gen) => (
                        <Accordion key={gen.id} className='generation-accordion'>
                            <AccordionSummary
                                expandIcon={<ExpandMoreIcon sx={{ color: '#fff' }} />}
                                aria-controls={`panel${gen.id}-content`}
                                id={`panel${gen.id}-header`}
                            >
                                <Typography component="span" sx={{ color: '#fff' }}>Generaci√≥n {gen.id}</Typography>
                            </AccordionSummary>
                            <AccordionDetails>
                                <Box className="pokemon-list">
                                    {gen.pokemon.map((pokemon, index) => (
                                        <Box key={`pokemon-${pokemon.name}-${index}`} 
                                            sx={{ 
                                                mb: 1, 
                                                cursor: 'pointer',
                                            }}
                                        >
                                            <Typography sx={{ color: '#fff' }}>{pokemon.name.charAt(0).toUpperCase() + pokemon.name.slice(1)}</Typography> 
                                        </Box>
                                    ))}
                                </Box>
                            </AccordionDetails>
                        </Accordion>
                    ))}
                </>
            )}
        </Box>
    );

}

export default GenerationTree