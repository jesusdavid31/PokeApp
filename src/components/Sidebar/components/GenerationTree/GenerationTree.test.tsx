import { render } from '@testing-library/react';
import GenerationTree from './GenerationTree';

// Mock de las funciones de react-router-dom
jest.mock('react-router-dom', () => ({
  useNavigate: jest.fn(),
  useLocation: jest.fn(),
}));

test('renders correctly', () => {
  const { container } = render(<GenerationTree />);
  expect(container).toMatchSnapshot();
});
