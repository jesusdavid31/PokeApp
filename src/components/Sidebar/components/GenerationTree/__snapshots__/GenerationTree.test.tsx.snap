// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`renders correctly 1`] = `
<div>
  <div
    class="generation-tree MuiBox-root css-3dhuck"
  >
    <div
      class="show-all-generations MuiBox-root css-0"
    >
      <div
        class="pokeball MuiBox-root css-0"
      />
      <p
        class="MuiTypography-root MuiTypography-body1 css-rizt0-MuiTypography-root"
      >
        PÃ¡gina Principal
      </p>
    </div>
  </div>
</div>
`;
